lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -133,21 +133,22 @@ class BlockedToMFMA : public mlir::RewritePattern {
 
   std::pair<int64_t, int64_t> chooseMfmaDimensions(triton::DotOp dot, int mfmaVersion) const {
     int64_t nonKDim = 32;
+    // number of consecutive matrix elements along k dim per thread
     int64_t kDim = -1;
```

Comment:
This is confusing. From `nonKDim`, kDim should mean the total number of element along the k dim for one mfma. Since you are choosing mfma dimensions here, it's better to revert the changes here and add `/ 2` later at line 211 and 214 when you construct the new dotOperandEncodings.